<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>790</width>
    <height>627</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Files to features tool</string>
  </property>
  <property name="dockNestingEnabled">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowNestedDocks|QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QSplitter" name="splitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
         <horstretch>0</horstretch>
         <verstretch>10</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <widget class="QWidget" name="widget" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>10</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QLineEdit" name="lineSearchComponents">
           <property name="placeholderText">
            <string>Search components and used filenames</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTreeView" name="treeComponents">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>160</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Add files by move items from the not-used-files view to a component and remove files by moving items from a component to the not-used-files view.</string>
           </property>
           <property name="dragEnabled">
            <bool>true</bool>
           </property>
           <property name="dragDropMode">
            <enum>QAbstractItemView::DragDrop</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="widgetComponentDetails" native="true">
           <layout class="QGridLayout" name="gridLayout">
            <property name="margin">
             <number>0</number>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Full path:</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QToolButton" name="btnChangeSubdir">
              <property name="text">
               <string>...</string>
              </property>
              <property name="icon">
               <iconset theme="folder">
                <normaloff/>
               </iconset>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="labelPath">
              <property name="text">
               <string>...</string>
              </property>
              <property name="textInteractionFlags">
               <set>Qt::TextBrowserInteraction</set>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_2">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Dependencies:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QToolButton" name="btnChangeDepends">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="labelDepends">
              <property name="text">
               <string>...</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Optional name:</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLineEdit" name="lineEdit"/>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="baseSize">
         <size>
          <width>100</width>
          <height>0</height>
         </size>
        </property>
        <property name="title">
         <string>List of files not used so far</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="lineSearchUnusedFiles">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="placeholderText">
            <string>Search not used files</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTreeView" name="treeFiles">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>100</width>
             <height>0</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Use drag and drop to move one or more files  to a component</string>
           </property>
           <property name="dragEnabled">
            <bool>true</bool>
           </property>
           <property name="dragDropMode">
            <enum>QAbstractItemView::DragDrop</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ExtendedSelection</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QGroupBox" name="groupProblems">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>50</height>
        </size>
       </property>
       <property name="title">
        <string> Problems detected:</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QListWidget" name="listProblems"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="bottomFrame" native="true">
      <property name="styleSheet">
       <string notr="true">#bottomFrame {
border: none;
background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
stop: 0 #a6a6a6, stop: 0.08 #7f7f7f,
stop: 0.39999 #717171, stop: 0.4 #626262,
stop: 0.9 #4c4c4c, stop: 1 #333333);
}
#bottomFrame QPushButton {
color: #333;
border: 2px solid #555;
border-radius: 7px;
padding: 2px;
padding-left:4px;
padding-right:4px;
background: qradialgradient(cx: 0.3, cy: -0.4,
fx: 0.3, fy: -0.4,
radius: 1.35, stop: 0 #fff, stop: 1 #888);
min-width: 80px;
}

#bottomFrame QPushButton:hover {
background: qradialgradient(cx: 0.3, cy: -0.4,
fx: 0.3, fy: -0.4,
radius: 1.35, stop: 0 #fff, stop: 1 #bbb);
}

#bottomFrame QPushButton:pressed {
background: qradialgradient(cx: 0.4, cy: -0.1,
fx: 0.4, fy: -0.1,
radius: 1.35, stop: 0 #fff, stop: 1 #ddd);
}

#bottomFrame QPushButton:checked {
background: qradialgradient(cx: 0.4, cy: -0.1,
fx: 0.4, fy: -0.1,
radius: 1.35, stop: 0 #fff, stop: 1 #ddd);
}

</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="margin">
        <number>2</number>
       </property>
       <item>
        <widget class="QPushButton" name="btnShowProblems">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show problems section.&lt;/p&gt;&lt;p&gt;The red number (if any) next to the label indicates how many proplems have been found.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Problems</string>
         </property>
         <property name="icon">
          <iconset theme="document-properties"/>
         </property>
         <property name="iconSize">
          <size>
           <width>24</width>
           <height>24</height>
          </size>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnShowComponentDetails">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show component details below the component tree.&lt;/p&gt;&lt;p&gt;Allows to edit the path, dependencies and optional name.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Component details</string>
         </property>
         <property name="icon">
          <iconset theme="document-properties"/>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnShowUnusedFiles">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Show unused-files section on the left side of the window&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Unused files</string>
         </property>
         <property name="icon">
          <iconset theme="document-properties"/>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>790</width>
     <height>27</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuComponents">
    <property name="title">
     <string>Components</string>
    </property>
    <addaction name="actionSynchronize_with_file_system"/>
    <addaction name="separator"/>
    <addaction name="actionAdd_component"/>
    <addaction name="actionRemove_selected_components"/>
    <addaction name="separator"/>
    <addaction name="actionExpand_all"/>
    <addaction name="actionExpand_all_missing_files_only"/>
    <addaction name="actionCollapse_all"/>
   </widget>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionClear_and_propose_component_structure"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuUnused_files">
    <property name="title">
     <string>Unused-files</string>
    </property>
    <addaction name="actionExpand_all_2"/>
    <addaction name="actionCollapse_all_2"/>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuComponents"/>
   <addaction name="menuUnused_files"/>
  </widget>
  <action name="actionDetectRemoved">
   <property name="icon">
    <iconset theme="edit-find-replace">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Detect moved and deleted files</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff/>
    </iconset>
   </property>
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="actionSynchronize_with_file_system">
   <property name="icon">
    <iconset theme="view-refresh"/>
   </property>
   <property name="text">
    <string>Synchronize with file system</string>
   </property>
   <property name="toolTip">
    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This will detect new files and removed files and update the problems section&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
   </property>
  </action>
  <action name="actionAdd_component">
   <property name="icon">
    <iconset theme="list-add"/>
   </property>
   <property name="text">
    <string>Add subcomponent</string>
   </property>
  </action>
  <action name="actionRemove_selected_components">
   <property name="icon">
    <iconset theme="list-remove"/>
   </property>
   <property name="text">
    <string>Remove selected components</string>
   </property>
  </action>
  <action name="actionExpand_all">
   <property name="text">
    <string>Expand all</string>
   </property>
  </action>
  <action name="actionCollapse_all">
   <property name="text">
    <string>Collapse all</string>
   </property>
  </action>
  <action name="actionExpand_all_2">
   <property name="text">
    <string>Expand all</string>
   </property>
  </action>
  <action name="actionCollapse_all_2">
   <property name="text">
    <string>Collapse all</string>
   </property>
  </action>
  <action name="actionExpand_all_missing_files_only">
   <property name="text">
    <string>Expand all missing files only</string>
   </property>
  </action>
  <action name="actionClear_and_propose_component_structure">
   <property name="icon">
    <iconset theme="document-new"/>
   </property>
   <property name="text">
    <string>Clear and propose component structure</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>btnShowProblems</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupProblems</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>35</x>
     <y>610</y>
    </hint>
    <hint type="destinationlabel">
     <x>185</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnShowUnusedFiles</sender>
   <signal>toggled(bool)</signal>
   <receiver>groupBox</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>613</y>
    </hint>
    <hint type="destinationlabel">
     <x>531</x>
     <y>364</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnShowComponentDetails</sender>
   <signal>toggled(bool)</signal>
   <receiver>widgetComponentDetails</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>211</x>
     <y>616</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>365</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCollapse_all</sender>
   <signal>triggered()</signal>
   <receiver>treeComponents</receiver>
   <slot>collapseAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExpand_all</sender>
   <signal>triggered()</signal>
   <receiver>treeComponents</receiver>
   <slot>expandAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionCollapse_all_2</sender>
   <signal>triggered()</signal>
   <receiver>treeFiles</receiver>
   <slot>collapseAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>652</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExpand_all_2</sender>
   <signal>triggered()</signal>
   <receiver>treeFiles</receiver>
   <slot>expandAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>652</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
